pkgs/by-name/ex/excalidraw_export/package.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  cairo,
+  pango,
+  pkg-config,
+}:
+
+buildNpmPackage rec {
+  pname = "excalidraw_export";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Timmmm";
+    repo = "excalidraw_export";
+    rev = "320c8be92f468e5e19564f83e37709b80afc0e46";
+    hash = "sha256-E5kYI8+hzObd2WNVBd0aQDKMH1Sns539loCQfClJs1Q=";
+  };
+
+  npmDepsHash = "sha256-5yec7BCi1c/e+y00TqxIeoazs49+WdKdfsskAqnVkFs=";
+
+  npmBuildScript = "compile";
+
+  buildInputs = [
+    cairo
+    pango
+  ];
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = {
+    description = "CLI to export Excalidraw drawings to SVG and PDF";
+    homepage = "https://github.com/Timmmm/excalidraw_export";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ venikx ];
+    mainProgram = "excalidraw_export";
```

I noticed the darwin builds are failing, and I don't have a darwin machine. I guess it would be better practice to limit the platforms?

```suggestion
    mainProgram = "excalidraw_export";
        platforms = [
      "aarch64-linux"
      "x86_64-linux"
    ];
```

---

That would work. If Darwin builds are supported upstream, but are broken locally (at the Nixpkgs level), setting `broken` to `stdenv.isDarwin` is better.

---

> That would work, but if Darwin builds are supported upstream, but are broken locally (at the Nixpkgs level), setting `broken` to `stdenv.isDarwin` is better.

I am not sure if I fully understand this. With upstream, do you mean the other dependencies original library? If yes, from what I understand, macOS should be supported by it, but I think it needs to be built differently. The last thing you mentioned, would look like this?

```suggestion
    mainProgram = "excalidraw_export";
    broken = stdenv.isDarwin; 
```

---

added this broken flag